projects/observability/src/shared/dashboard/data/graphql/explore/explore-cartesian-data-source.model.ts

```diff
@@ -19,7 +32,51 @@ import { ExploreResult } from './explore-result';
   type: 'explore-cartesian-data-source'
 })
 export class ExploreCartesianDataSourceModel extends GraphQlDataSourceModel<CartesianDataFetcher<ExplorerData>> {
-  public request?: ExploreVisualizationRequest;
+  @ModelProperty({
+    key: 'context',
+    displayName: 'Context',
+    type: STRING_PROPERTY.type
+  })
+  public context?: string;
+
+  @ModelProperty({
+    key: 'metric-explore-series',
+    displayName: 'Metric explore series',
+    type: ARRAY_PROPERTY.type,
+    required: true
+  })
+  public metricExploreSeries: ExploreSeries[] = [];
+
+  @ModelProperty({
+    key: 'group-by',
```

Comment:
I'm confused - don't these each apply to individual series? But when grouping we also don't know the series until the results are retrieved, which is why we've never made a data source for them other than the one dynamically populated with a result object

Reply:
Yeah. So if we have group by, one series will be generated for each group. I am trying to make a DS for such a case.

Reply:
Going to convert this into an abstract class